omen2/object.py

```diff
@@ -339,13 +350,22 @@ def __enter__(self):
         return self
 
     def __exit__(self, typ, val, ex):
-        # unbound objects aren't locked, and don't need the with: protocol
+        """Finished with write, call commit or not, based on exception."""
         if not self.__meta or not self.__meta.locked:
+            # unbound objects aren't locked, and don't need the with: protocol
             return
 
         try:
             if not typ:
-                self._commit()
+                # see self._atomic_update() for why this works
+                saved = self.__dict__  # pylint: disable=access-member-before-definition
+                try:
+                    self._commit()
+                except Exception:
+                    self.__dict__ = saved
+                    raise
+        except OmenRollbackError:
```

Comment:
What I meant with this is that the `with` user would throw `OmenRollbackError`, which would come through in `typ`, and then we could return `True` here to suppress

Reply:
yeah, i knew what you meant, i just wasnt thinking.  will add a test

Reply:
updated